src/m365/outlook/commands/room/room-list.ts

```diff
@@ -0,0 +1,66 @@
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  roomlistEmail?: string;
+}
+
+class OutlookRoomListCommand extends GraphCommand {
+  public get name(): string {
+    return commands.ROOM_LIST;
+  }
+
+  public get description(): string {
+    return 'Get a collection of all available rooms';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.roomlistEmail = typeof args.options.roomlistEmail !== 'undefined';
+    return telemetryProps;
+  }
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'displayName', 'phone', 'emailAddress'];
+  }
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    let endpoint: string = `${this.resource}/v1.0/places/microsoft.graph.room`;
+    if (args.options.roomlistEmail) {
+      endpoint = `${this.resource}/v1.0/places/${args.options.roomlistEmail}/microsoft.graph.roomlist/rooms`;
+    }
+    const requestOptions: any = {
+      url: endpoint,
+      headers: {
+        accept: 'application/json;odata.metadata=none',
+        'content-type': 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    request.get(requestOptions)
```

Comment:
After you implemented GraphItemsListCommand you can use the getAllItems method

Reply:
used getAllItems method.